include/gadget/macros.h

```diff
@@ -11,4 +11,9 @@
 #define GADGET_TRACE_MAP(name) \
 	const void * gadget_trace_map_##name __attribute__((unused));
 
+// GADGET_PARAM is used to indicate that a given variable is used as a parameter for a given eBPF
+// Users of Inspektor Gadget can set these values from userspace
+#define GADGET_PARAM(name) \
+	const void * gadget_param_##name __attribute__((unused));
```

Comment:
I think the idea here is to have the metadata as the single source of truth.

Additionally if we would "mark" every `global const volatile` variable as a parameter all variables from the includes will be also shown for the gadget.
For example we include the `mntns_filter.h` header which would in turn expose the `gadget_filter_by_mntns` variable directly to the user. For that variable we have another specific flag since we need to do more for filtering by mntns than setting the variable to true.
But on the other hand we could just filter out all `global const volatile` variables which have a `gadget_` prefix.

In the end for me the `GADGET_PARAM` macro is more useful.

Other thoughts?

Reply:
> Note that, the metadata would still be used to set the parameters even if we do not use macro.

Correct and this is intended and what I also like